matches/matches.go

```diff
@@ -2,19 +2,161 @@ package main
 
 import (
 	"encoding/json"
+	"fmt"
 	"log"
 	"net/http"
+	"strconv"
 
+	"github.com/TempleEight/spec-golang/matches/dao"
+	valid "github.com/asaskevich/govalidator"
 	"github.com/gorilla/mux"
 )
 
 func main() {
 	r := mux.NewRouter()
-	r.HandleFunc("/matches", matchGetHandler).Methods(http.MethodGet)
+	r.HandleFunc("/matches/{id}", matchListHandler).Methods(http.MethodGet)
+	r.HandleFunc("/match/{id}", matchGetHandler).Methods(http.MethodGet)
+	r.HandleFunc("/match", matchCreateHandler).Methods(http.MethodPost)
+	r.HandleFunc("/match", matchDeleteHandler).Methods(http.MethodDelete)
+	r.HandleFunc("/match", matchUpdateHandler).Methods(http.MethodPatch)
 
 	log.Fatal(http.ListenAndServe(":81", r))
 }
 
 func matchGetHandler(w http.ResponseWriter, r *http.Request) {
-	json.NewEncoder(w).Encode("Hello, World")
+	matchIDStr := mux.Vars(r)["id"]
+	if len(matchIDStr) == 0 {
+		http.Error(w, CreateErrorJSON("No match ID provided"), http.StatusBadRequest)
+		return
+	}
+
+	matchID, err := strconv.ParseInt(matchIDStr, 10, 64)
+	if err != nil {
+		http.Error(w, CreateErrorJSON("Invalid match ID provided"), http.StatusBadRequest)
+		return
+	}
+
+	match, err := dao.GetMatch(matchID)
+	if err != nil {
+		switch err.(type) {
+		case dao.ErrMatchNotFound:
+			http.Error(w, CreateErrorJSON(err.Error()), http.StatusNotFound)
+		default:
+			errMsg := CreateErrorJSON(fmt.Sprintf("Something went wrong: %s", err.Error()))
+			http.Error(w, errMsg, http.StatusInternalServerError)
+		}
+		return
+	}
+
+	json.NewEncoder(w).Encode(match)
+}
+
+func matchListHandler(w http.ResponseWriter, r *http.Request) {
+	userOneStr := mux.Vars(r)["id"]
```

Comment:
Perhaps rename to just `userStr` since can be userOne or userTwo in the match, and similarly throughout the rest of this func